node/network/collator-protocol/src/validator_side/mod.rs

```diff
@@ -960,6 +957,19 @@ where
 				"CollationFetchingRequest message is not expected on the validator side of the protocol",
 			);
 		}
+		Seconded(parent, stmt) => {
+			if let Some(collation_event) = state.seconded.remove(&parent) {
+				let (collator_id, pending_collation) = collation_event;
+				let PendingCollation { relay_parent, peer_id, .. } = pending_collation;
+				note_good_collation(ctx, &state.peer_data, collator_id).await;
+				notify_collation_seconded(ctx, peer_id, relay_parent, stmt).await;
+			}
+		}
+		Invalid(parent, _) => {
```

Comment:
should we check the candidate is the same we seconded before removing the collation and reporting the collator?